image/tarexport/tarexport.go

```diff
@@ -24,10 +26,12 @@ type manifestItem struct {
 }
 
 type tarexporter struct {
-	is             image.Store
-	lss            layer.Store
-	rs             refstore.Store
-	loggerImgEvent LogImageEvent
+	is              image.Store
+	lss             layer.Store
+	rs              refstore.Store
+	loggerImgEvent  LogImageEvent
+	platform        *platforms.Platform
```

Comment:
Not sure platform is needed here since we have the matcher.

Reply:
It's used in error message: https://github.com/moby/moby/blob/c25d20beb91db1a4811c0529eb8e5eba7dff5c7b/image/tarexport/save.go#L171

Reply:
Yeah, would be great if matchers would expose more information about what they're matching and/or perhaps return an error instead of only have a boolean.